### PR TITLE
Chainer Unified encoder

### PR DESCRIPTION
**Pull request by Fhrozen**
Transcribing pytorch modification 
Attending at #624

### CODE REVIEW

**Comment by Fhrozen** [comment]
@sw005320 , Tested all encoders in AN4. Executed without problem.
Just finished the additional fixes, tried to clean up the configuration code for RNN. The code is quite similar to #624 proposed by @JaejinCho .
Let me know if there is any additional code to fix, or to discuss with @JaejinCho.

**Comment by sw005320** [comment]
Can you prepare a test script for the unidirectional RNN?

**Comment by Fhrozen** [comment]
Are the added test correct? should I add more? I tried to gather them by backend.

**Comment by sw005320** [comment]
@JaejinCho, could you also review this PR?

**Comment by Fhrozen** [comment]
I modified the tests according as @sw005320 proposed. Let me know if this fits.
I also adding a fix about travis. Seems that the files cannot be downloaded from drive without confirmation.
I used this [reference](https://www.matthuisman.nz/2019/01/download-google-drive-files-wget-curl.html)

**Comment by sw005320** [comment]
Can you add test examples of unidirectional lstm/gru with VGG (e.g., `vgglstm` and `vgglstmp`) for both chainer and pytorch?

**Comment by Fhrozen** [comment]
@sw005320 , I am not sure why travis is giving error, but when I test pytest in my env with conda I have no problem. is it required to add the compatibility test?. This two are the only problem.

**Comment by sw005320** [comment]
We noticed this issue. You can ignore them for now.
@kan-bayashi, do you have any idea?

**Comment by kan-bayashi** [comment]
I made PR to fix this issue (#649).
Download from google drive is a little bit difficult...

File: espnet/nets/chainer_backend/encoders.py [hunk 1]
```diff
@@ -13,27 +13,47 @@
...
+    :param np.ndarray subsample: list of subsampling numbers
+    :param float dropout: dropout rate
+    :param str typ: The RNN type
+    """
+
+    def __init__(self, idim, elayers, cdim, hdim, subsample, dropout, typ="blstm"):
+        super(RNNP, self).__init__()
+        bidir = typ[0] == "b"
+        if bidir:
+            linear_conn = L.NStepBiLSTM if "lstm" in typ else L.NStepBiGRU
```
[the remark this answers is not in the document]
Comment (Fhrozen) [reply espnet/nets/chainer_backend/encoders.py]: ty for the revision, I just modified as proposed. =)

**Comment by sw005320** [comment]
thanks!

**Comment by Fhrozen** [comment]
let me know about that, but since vgg test are included (+ lstm/gru) should not that be enough??.

**Pull request merged**